### ISSUE TITLE
Map pack above a size should be cuted in smaller area

### ISSUE THREAD

**Issue by cm-t** [fransschreuder/libosmscout#13]
Hi

Today map packs are country based. 
It should be as seen on OSM-And (the libre android app): Country for small pack, country's region for big pack:

For example, I should be able to choose France/Bretagne or uk/scotland.

I know there is an issue about Sdcard and apparmor policy. But right now we use the internal memory, let's deal with it:

I can't use the app at all, i am in France/île-de-France, i have all my personnal files on the sdcard, but the system still tells me i have only less than 2Go on remaining on the internal memory (bq aquaris E4.5). On the map selector, I have to pick the full France map pack. It sizes like 3Go. Can't use the map for my country.

I will go in scotland this summer, I have to download the full UK map pack that migh fit or not on the phone (it might download the things, but it won't be able to have the archive + extracted files if it is a need for the app to extract).

TL;DR: Make smaller map pack when it's above something like 300Mo to be sure the app can run for any location at minimal.

**Comment by cm-t** [comment]
(I don't have OSMAnd right now bacause I don't have anymore an ASOP based phone, copy the same sizes, maybe the maxsize is above or below 300Mo)

**Comment by fransschreuder** [comment]
I have converted all the countries in the world, if you want me to convert sub-regions as well, I will be busy forever. There are possibilities and tools to import the maps of your choice, using a desktop PC.
If you want to use France on your sd-card you can get the app from open.uappexplorer.com.

**Comment by cm-t** [comment]
You looks to agree someone should cut the size, if you don't have time, just leave the issue open to let know it is needed (we never know, maybe someone will read this and pull request or something, freesoftware)

Note that I just said that only big pack should be cuted in sub regionpack. France is a well example.

Did you see how osmAnd works? Maybe you couls fetch their packs, or clone their repos.(for the good example of france, they have a french package in a hand, but no heavy zoom, and region package in the other hand, for detailled route on local area/zoom if my memory serves me well).

**Comment by cm-t** [comment]
I have just installed the "open" version.
I am at 48% of the "OverAll progress", and this is now for 1 hour, So I suppose it will be 2 hours to have the French package.

I think this is a major issue: will user have to wait 2 hours to use the apps everytimes there is an update ?

Please open the issue, even marked as enhancement, even if you won't do it personally (I so understand you are missing the time to do it). I strongly beleive this is a critical feature.

**Comment by Theodin** [comment]
OSMScout uses a different library to display the mapy that OSMAnd so they cannot be reused. Having used OSMAnd, I know how practical their maps are, but if there is no way to generate the maps in an automated way, I can understand fransschreuder if he wants to save his time for more important features.
I downloaded my region from http://www.geofabrik.de/ as +.OSM file and generated the map files myself for the smaller region where I live which worked well after some help from fransschreuder.

**Comment by fransschreuder** [comment]
Thanks Theodin for your explanation. 
cm-t: if you can't wait for your map to download, you should probably search for a better internet connection or create a smaller map region yourself. 
I will work on a way to automate map imports, but for now I want to use the regions that geofabrik.de provides, they don't provide for example subregions of France. 
I am not reopening the issue because maps are not integrated in the app, and this issue just generates noise and slows down the development.

**Comment by Theodin** [comment]
They do:
http://download.geofabrik.de/europe/france.html

**Comment by fransschreuder** [comment]
Ok, but then again, I must think of a way to select them, if you put all subregions of all countries into one list, it will be to big to handle from within the app.

**Comment by fransschreuder** [comment]
...later

**Comment by Theodin** [comment]
Yes. Maybe with subpages like the geofabrik does? But that would require changes to you current one-page-setup...

**Comment by fransschreuder** [comment]
Exactly... But as it seems now, I only see subpages for Germany and France right? I could simply upload them as france_\* and germany_*

**Comment by fransschreuder** [comment]
Ok, I didn't know thos subregions of France and Germany were available... I have just uploaded france_\* and germany_*. Have fun!

**Comment by Theodin** [comment]
Thats great. France and Germany are by far the biggest files so that probably why the Geofabrik serves smaller sections.
